iiif_prezi3/helpers/add_thumbnail.py

```diff
@@ -21,5 +22,75 @@ def add_thumbnail(self, image_url, **kwargs):
         self.thumbnail.append(new_thumbnail)
         return new_thumbnail
 
+    def create_thumbnail_from_iiif(self, url, preferred_width=500, **kwargs):
+        """Adds an image thumbnail to a manifest or canvas based on a IIIF service.
+
+        Args:
+            url (str): An HTTP URL which points at a IIIF Image response.
+            preferred_width (int, optional): the preferred width of the thumbnail.
+            **kwargs (): see ResourceItem.
+
+        Returns:
+            new_thumbnail (ResourceItem): The updated list of thumbnails, including the newly-created one.
+        """
+        image_response = ResourceItem(id=url, type='Image')
+        image_info = image_response.set_hwd_from_iiif(url)
+        context = image_info.get('@context', '')
+
+        if 'sizes' in image_info:
+            best_fit_size = max(
+                (size for size in image_info['sizes'] if size["width"] <= preferred_width),
```

Comment:
We should get a bigger image as this can be shrinked with css and not loose quality. If we go smaller css will enlarge it and it may get pixilated.